databuilder/backends/base.py

```diff
@@ -39,67 +39,51 @@ def get_table_expression(self, table_name, schema):
         Raises:
             ValueError: If unknown table passed in
         """
-        # TODO: We currently ignore the `schema` argument here. But I think we should
-        # move towards having the supplied schema define the column types or, if not
-        # that, then we should validate that the types match.
-        return self.tables[table_name].get_expression(table_name)
+        return self.tables[table_name].get_expression(table_name, schema)
 
 
 class SQLTable:
     def learn_patient_join(self, source):
-        raise NotImplementedError()
-
-    def _make_columns(self):
-        return [
-            self._make_column(name, column) for name, column in self.columns.items()
-        ]
-
-    def _make_column(self, name, column):
-        source = column.source or name
-        type_ = TYPES_BY_NAME[column.type].value
-        sql_column = sqlalchemy.Column(source, type_, key=name)
-        return sql_column
+        self.patient_join_column = source
 
 
 class MappedTable(SQLTable):
     def __init__(self, source, columns, schema=None):
-        self.source = source
-        self.columns = columns
-        self._schema = schema
-
-    def learn_patient_join(self, source):
-        if "patient_id" not in self.columns:
-            self.columns["patient_id"] = Column("integer", source)
+        self.source_table_name = source
+        self.column_map = columns
+        # Not to be confused with the schema which defines the column names and types
+        self.db_schema_name = schema
```

Comment:
I'm so cross about this duplicated standard usage in SQL-land.

Reply:
I don't see the problem. It's either the name for a collection of tables that's sort of like a database but not a database. Or it's a list of those tables. Or it's a list of the _columns_ of those tables. Or it's either of those things, but in ehrQL rather than SQL.

Reply:
:grinning: